src/xrc/src/lib.rs

```diff
@@ -486,6 +504,23 @@ impl QueriedExchangeRate {
     }
 }
 
+/// The function validates the rates in the [QueriedExchangeRate] struct.
+pub(crate) fn validate(
+    rate: QueriedExchangeRate,
+) -> Result<QueriedExchangeRate, ExchangeRateError> {
+    if median(&rate.rates) == 0 {
+        return Err(ExchangeRateError::Other(OtherError {
+            code: INVALID_RATE_ERROR_CODE,
+            description: INVALID_RATE_ERROR_MESSAGE.to_string(),
+        }));
+    }
+    if rate.is_valid() {
+        Ok(rate)
+    } else {
+        Err(ExchangeRateError::InconsistentRatesReceived)
+    }
+}
+
```

Comment:
Could we combine `is_valid` and this `validate` function? Perhaps something like....

```rust
impl QueriedExchangeRate {
    fn validate(self) -> Result<Self, ExchangeRateError> {
       // validation logic here
   }
}
```

Reply:
Yeah, I had this idea as well.
Let me try it out!

Reply:
Done! Please take another look!